keras/layers/core.py

```diff
@@ -1213,3 +1213,31 @@ def get_config(self):
                   'input_length': self.input_length}
         base_config = super(TimeDistributedDense, self).get_config()
         return dict(list(base_config.items()) + list(config.items()))
+
+
+class GradientReversal(Layer):
+    '''
+    Flip the sign of gradient during training.
+
+    # Arguments:
+        hp_lambda: Scalar to multiply the flipped gradient.
+    '''
+    def __init__(self, hp_lambda, **kwargs):
+        super(GradientReversal, self).__init__(**kwargs)
+        self.hp_lambda = K.variable(hp_lambda)
+        self.supports_masking = False
+        self.op = K.ReverseGradient(self.hp_lambda)
+
+    def build(self, input_shape):
+        self.trainable_weights = []
+
+    def call(self, x, mask=None):
+        return self.op(x)
```

Comment:
If all you do with `ReverseGradient` is call it, why should it be a class? Everything in the backend is a function.